### PR TITLE
Refresh ECS agent installation

### PR DESCRIPTION
**Pull request by jmsanders**
Our ECS installation docs are kind of a weird spot because our CloudFormation template is written to get you from 0 to 1 as quickly as possible. We leave configuration via CloudFormation variables pretty minimal because the customization story just isn't as strong as other infrastructure-as-code tools.

Additionally, whereas with K8s we can be pretty hand-wavy and say "make sure you have a K8s cluster already set up," with ECS we're forced to make some pre-requisite decisions (like network configuration, IAM policies, etc.) on our users' behalf. This creates an uncomfortable tension where we don't want to be too proscriptive on these things, but we also don't want the burden of supporting customization for all possible security postures.

I don't really want to turn this PR into a referrendum on CloudFormation
- I agree that it'd be useful if we had other ways of launching a more customizable ECS agent (or exampled, a Terraform module. Or de-coupling dagster.yaml from the CloudFormation template itself). But I consider that out of scope for this documentation refresh.

I've chosen to collapse the two recommended guides (with and without an existing VPC) into one. And the guide itself is pretty minimal - once you click the launch button, the CloudFormation wizard should guide you through the rest.

### CODE REVIEW

File: docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+title: "Running Dagster+ agents on Amazon ECS"
+displayed_sidebar: "dagsterPlus"
+sidebar_position: 11
+sidebar_label: "Amazon ECS"
+---
+
+# Running Dagster+ agents on Amazon ECS
+
+This page provides instructions for running the [Dagster+ agent](dagster-plus/getting-started/whats-dagster-plus#Agents) on an [Amazon ECS](https://aws.amazon.com/ecs/) cluster.
+
```
Comment (cnolanminich) [inline docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md:11]: I think the existing prerequisites sections for ECS agents [in a new VPC](https://docs.dagster.io/dagster-plus/deployment/agents/amazon-ecs/creating-ecs-agent-new-vpc#prerequisites) or [an existing VPC](https://docs.dagster.io/dagster-plus/deployment/agents/amazon-ecs/creating-ecs-agent-existing-vpc#prerequisites) are helpful, I would want them in this guide as well

**Review by cnolanminich — COMMENTED** [review]
@jmsanders just a few points of clarification comparing this against the existing docs:

* are you suggesting we remove the current docs section on [manually provisioning an ECS agent](https://docs.dagster.io/dagster-plus/deployment/agents/amazon-ecs/manually-provisioning-ecs-agent)
* are you assuming that generating the token step is already completed? A link to that section might be helpful in that case.

**Review by mlarose — COMMENTED** [review]
there's a ton of important information in these pages and we probably should link to this:

- https://docs.dagster.io/dagster-plus/deployment/agents/amazon-ecs/creating-ecs-agent-new-vpc
- https://docs.dagster.io/dagster-plus/deployment/agents/amazon-ecs/creating-ecs-agent-existing-vpc
- https://docs.dagster.io/dagster-plus/deployment/agents/amazon-ecs/manually-provisioning-ecs-agent

**Comment by mlarose** [comment]
Also if anyone else ends up expanding on this, there is this draft I never got around to completing that goes into the dissection of what's needed to roll your own and explains how the whole thing works:

- https://www.notion.so/dagster/Understanding-AWS-ECS-Dagster-deployments-f55d8c166ec0466eb3361186cb13c2cd

File: docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+## Installation
+
+You can install the Dagster+ agent using one of our [CloudFormation](https://aws.amazon.com/cloudformation/) templates.
+
+To install into a new [VPC](https://aws.amazon.com/vpc/):
+
+[<img src="https://s3.amazonaws.com/cloudformation-examples/cloudformation-launch-stack.png"/>](https://console.aws.amazon.com/cloudformation/home#/stacks/create/review?templateURL=https://s3.amazonaws.com/dagster.cloud/cloudformation/ecs-agent.yaml)
+
+For more flexibility, you can install into an existing VPC:
```
Comment (mlarose) [inline docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md:20]: i'd explain what's the added flexibility here.

```suggestion
For more flexibility, you can install into an existing VPC:

Reasons you might want to do this is to integrate within pre-existing network, adopt a network topology that differs from the recommended patterns, or have security, compliance or other operational constraints that are best supported by a different configuration.
```

File: docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+[<img src="https://s3.amazonaws.com/cloudformation-examples/cloudformation-launch-stack.png"/>](https://console.aws.amazon.com/cloudformation/home#/stacks/create/review?templateURL=https://s3.amazonaws.com/dagster.cloud/cloudformation/ecs-agent-vpc.yaml)
+
+For even more flexibility, you can edit these templates or use them as a guide for managing your own infrastructure.
```
Comment (jmsanders) [inline docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md:24]: > are you suggesting we remove the current docs section on [manually provisioning an ECS agent](https://docs.dagster.io/dagster-plus/deployment/agents/amazon-ecs/manually-provisioning-ecs-agent)

🤷

I don't have a strong preference one way or another. @erinkcochran87 did you imagine these new guides would replace this old documentation or just be a lighter-weight version of them? 

My sense from reading that guide is that there's not any information you wouldn't already be able to pick up from opening the CloudFormation template and looking at it. It feels like not enough information for someone who doesn't already have pretty deep context to roll their own. But you probably have a better sense than I do about how often these guides get referenced in installations.

Comment (cnolanminich) [reply docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md:24]: I have never seen someone use that one -- I think they use one of the two cloud formation templates but I'm curious what @slopp thinks

Comment (slopp) [reply docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md:24]: The manual guide would be used by teams who aren't using Cloud Formation. Often, these teams are fairly competent and go from our CF to their own thing, I think often the information they need is more of what Mathieu has in his notion guide.

File: docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+title: "Running Dagster+ agents on Amazon ECS"
+displayed_sidebar: "dagsterPlus"
+sidebar_position: 11
+sidebar_label: "Amazon ECS"
+---
+
+# Running Dagster+ agents on Amazon ECS
```
Comment (slopp) [inline docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md:8]: Should we give a pointer to getting an agent token here?

File: docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md (hunk 3 above)
Comment (slopp) [inline docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/amazon-ecs.md:24]: I would add to this: 
- how they can know they were successful 
- next steps for ECS specific config

(Basically my same feedback from the k8s one)

**Review by slopp — COMMENTED** [review]
This lgtm if we assume the old guides are sticking around as-is

Basically, I view this doc as completing the guide-per-agent matrix, even if this one is really just a pointer at different cloud formation templates/docs.

If we're replacing those old docs... then I think we'd want to do more here. Specifically the areas I find prospects struggle with the most is:

- forgetting to enable branch deployments in the template
- understanding that our template manages the dagster.yaml file 
- often they just want to see the template without clicking through to the AWS console... a plaintext version of it might be useful 
- the conceptual understanding of:
   - what role do I need?
   - what network access do I need?

**Comment by erinkcochran87** [comment]
> did you imagine these new guides would replace this old documentation or just be a lighter-weight version of them?

The intent was to replace them, yes. But given the comments here and Mathieu's draft, it seems like there's more to do than just hashing out this page.

At a minimum, this guide, focused on CloudFormation, should include:

- The prereqs to use it, including Dagster+ permissions and what's required in AWS, such as permissions and networking
- A link to instructions on creating a Dagster+ agent token
- Info about some of the common configuration options to set, so users don't miss them. Ex: Branch Deployments

It looks like the template is currently in the internal repo, so we can't link to it. Maybe we could set up a process that copies a version of it to the docs so it's linkable?

@cnolanminich @slopp - Would teams who use CloudFormation still be interested in some of the info in Mathieu's draft? I'm wondering if everything in there can act as an explainer page for how ECS generally works with Dagster+, and then we still have a manual provisioning guide that's step-by-step. That would allow us to cross-link to useful info without polluting actionable steps, such as linking to more info about IAM permissions from guide prereqs.

Let me know if I'm way off here - I'd like to talk through this more either way after Docathon, as it's fairly complex.

**Review by gibsondan — CHANGES_REQUESTED** [review]
sound like there are some potential next steps here for reconciling this with the existing guides

**Pull request closed without merging**